lib/core/src/Cardano/Wallet.hs

```diff
@@ -1621,13 +1644,105 @@ forgetTx
 forgetTx ctx wid tid = db & \DBLayer{..} -> do
     mapExceptT atomically $ removePendingOrExpiredTx (PrimaryKey wid) tid
   where
-    db = ctx ^. dbLayer @s @k
+    db = ctx ^. dbLayer @IO @s @k
+
+-- | Given a LocalTxSubmission record, calculate the slot when it should be
+-- retried next.
+--
+-- The current implementation is really basic. Retry about once 10 blocks.
+scheduleLocalTxSubmission
+    :: Word64
+    -> SlottingParameters
+    -> LocalTxSubmissionStatus tx
+    -> SlotNo
+scheduleLocalTxSubmission numBlocks sp st =
+    (st ^. #latestSubmission) + numSlots
+  where
+    numSlots = SlotNo (ceiling (fromIntegral numBlocks / f))
+    ActiveSlotCoefficient f = getActiveSlotCoefficient sp
+
+-- | Continuous process which monitors the chain tip and retries submission of
+-- pending transactions as the chain lengthens.
+--
+-- The current default is to resubmit any pending transaction about every 10
+-- blocks.
+--
+-- Regardless of the frequency of chain updates, this function won't re-query
+-- the database faster than every 1000ms.
+--
+-- This only exits if the network layer 'watchNodeTip' function exits.
+runLocalTxSubmissionPool
+    :: forall ctx s k m.
+        ( MonadUnliftIO m
+        , MonadMonotonicTime m
+        , HasLogger WalletLog ctx
+        , HasNetworkLayer m ctx
+        , HasDBLayer m s k ctx
+        )
+    => ctx
+    -> WalletId
+    -> m ()
+runLocalTxSubmissionPool = runLocalTxSubmissionPool' @ctx @s @k @m 1 10
+
+-- | A more configurable (i.e. testable) version of 'runLocalTxSubmissionPool''.
+runLocalTxSubmissionPool'
+    :: forall ctx s k m.
+        ( MonadUnliftIO m
+        , MonadMonotonicTime m
+        , HasLogger WalletLog ctx
+        , HasNetworkLayer m ctx
+        , HasDBLayer m s k ctx
+        )
+    => DiffTime -- ^ Minimum time between updates
+    -> Word64 -- ^ Retry approximately every so many blocks
```

Comment:
(suggestion) Consider making those a record to make things clearer at call and consumer sites:

```hs
data LocalTxSubmissionConfig = LocalTxSubmissionConfig
    { rateLimit :: DiffTime
    , blockInterval :: Word64
    }
```

Reply:
Good idea, fixed.